pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -393,54 +397,84 @@ func (r *rbac) createResources(ctx context.Context) error {
 
-		// We use a namespace scoped Role when SCC annotation is present, and
-		// a cluster scoped ClusterRole when the default SCC is used
-		roleRef := r.getSCCRoleInNamespace(&ns)
-		if err := r.ensurePipelinesSCCRoleBinding(ctx, sa, roleRef); err != nil {
-			return fmt.Errorf("failed to create Pipeline Scc Role Binding in namespace %s, %w", ns.Name, err)
-		}
+	var hasError bool
 
-		if err := r.ensureRoleBindings(ctx, sa); err != nil {
-			return err
-		}
+	logger.Infof("ensure ca bundle configmap in this namespace")
+	if err := r.ensureCABundles(ctx, &ns); err != nil {
+		hasError = true
+		logger.Errorf("failed to ensure ca bundles presence in this namespace, %w", err)
+	}
 
-		if err := r.ensureClusterRoleBindings(ctx, sa); err != nil {
-			return err
-		}
+	logger.Infof("ensures default pipelines service account in this namespace")
+	sa, err := r.ensureSA(ctx, &ns)
+	if err != nil {
+		hasError = true
+		logger.Errorf("failed to ensure default SA in this namespace %w", err)
+	}
+
+	// If "operator.tekton.dev/scc" exists in the namespace, then bind
+	// that SCC to the SA
+	err = r.handleSCCInNamespace(ctx, &ns)
+	if err != nil {
+		hasError = true
+		logger.Errorf("failed to bind scc to this namespace %w", err)
+	}
+
+	// We use a namespace scoped Role when SCC annotation is present, and
+	// a cluster scoped ClusterRole when the default SCC is used
+	roleRef := r.getSCCRoleInNamespace(&ns)
+	if err := r.ensurePipelinesSCCRoleBinding(ctx, sa, roleRef); err != nil {
+		hasError = true
+		logger.Errorf("failed to create Pipeline Scc Role Binding in this namespace %w", err)
+	}
 
-		// Add `openshift-pipelines.tekton.dev/namespace-reconcile-version` label to namespace
-		// so that rbac won't loop on it again
+	if err := r.ensureRoleBindings(ctx, sa); err != nil {
+		hasError = true
+		logger.Errorf("failed to create rolebinding in this namespace %w", err)
+	}
+
+	if err := r.ensureClusterRoleBindings(ctx, sa); err != nil {
+		hasError = true
+		logger.Errorf("failed to create clusterrolebinding in this namespace %w", err)
+	}
+
+	// if No error add `openshift-pipelines.tekton.dev/namespace-reconcile-version` label to namespace
+	// so that rbac won't loop on it again
+	if !hasError {
+		logger.Infof("namespace %s sucessfully reconciled. Adding label namespace-reconcile-version to mark it as reconciled", ns.Name)
 		nsLabels := ns.GetLabels()
 		if len(nsLabels) == 0 {
 			nsLabels = map[string]string{}
 		}
 		nsLabels[namespaceVersionLabel] = r.version
 		ns.SetLabels(nsLabels)
 		if _, err := r.kubeClientSet.CoreV1().Namespaces().Update(ctx, &ns, metav1.UpdateOptions{}); err != nil {
-			return fmt.Errorf("failed to update namespace %s with label %s, %w", ns.Name, namespaceVersionLabel, err)
+			logger.Errorf("failed to update namespace %s with label %s, %w", ns.Name, namespaceVersionLabel, err)
 		}
+	} else {
+		logger.Errorf("failed to reconcile namespace %s, %w", ns.Name, err)
```

Comment:
just a small thought like, dont we need to requeue the request or something to process it again?

Reply:
true, i ve added an error chan that capture the error and return it to main createResource. createResource returns the error to the caller, and request is then requeued

Reply:
aah ohk fine